logdna/resource_category.go

```diff
@@ -0,0 +1,165 @@
+package logdna
+
+import (
+  "context"
+  "encoding/json"
+  "fmt"
+  "log"
+
+  "github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+  "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCategoryCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+  var diags diag.Diagnostics
+  pc := m.(*providerConfig)  
+
+  categoryType := d.Get("type").(string)
```

Comment:
Creating a resource should allow for `name`, but it's not sent here.  I see that `CreateRequestBody` is filling in `name`, but not `type`.  That's super confusing since name is required, IMO.  It makes it look like things are missing here.

I would move the above line to `CreateRequestBody`, and then to supply it below, pull it from the populated `category` structure below.

Reply:
@darinspivey The type is not part of a request body, that why I set just name in `CreateRequestBody`, type I need just to generate a url. I wanted to avoid sending type field in request body.

Reply:
Hmm, I get it, but that's confusing since the TF schema has `type` and `name` alongside each other.  Add a comment to clarify and delete the commented out `Type` in `request_types.go`